rc/filetype/markdown.kak

```diff
@@ -59,9 +59,20 @@ add-highlighter shared/markdown/codeblock region -match-capture \
     ^(\h*)```\h*$ \
     fill meta
 
-add-highlighter shared/markdown/listblock region ^\h*[-*]\s ^\h*((?=[-*])|$) group
-add-highlighter shared/markdown/listblock/ ref markdown/inline
-add-highlighter shared/markdown/listblock/marker regex ^\h*([-*])\s 1:bullet
+add-highlighter shared/markdown/listblock region ^\h*[-*]\s ^\h*((?=[-*])|$) regions
+add-highlighter shared/markdown/listblock/g default-region group
+add-highlighter shared/markdown/listblock/g/ ref markdown/inline
+add-highlighter shared/markdown/listblock/g/marker regex ^\h*([-*])\s 1:bullet
+
+# https://spec.commonmark.org/0.29/#link-destination
+# This repetition is not pretty but shell escaping is worse
+add-highlighter shared/markdown/angle_bracket_url region (?<=<)([a-z]+://|(mailto|magnet|xmpp):) (?!\\).(?=>)|\n fill link
+add-highlighter shared/markdown/url region -recurse \( ([a-z]+://|(mailto|magnet|xmpp):) (?!\\).(?=\))|\s fill link
+add-highlighter shared/markdown/listblock/angle_bracket_url region (?<=<)([a-z]+://|(mailto|magnet|xmpp):) (?!\\).(?=>)|\n fill link
+add-highlighter shared/markdown/listblock/url region -recurse \( ([a-z]+://|(mailto|magnet|xmpp):) (?!\\).(?=\))|\s fill link
+
+require-module html
+add-highlighter shared/markdown/tag region </?(a|abbr|acronym|address|applet|area|article|aside|audio|b|base|basefont|bdi|bdo|big|blockquote|body|br|button|canvas|caption|center|cite|code|col|colgroup|data|datalist|dd|del|details|dfn|dialog|dir|div|dl|dt|em|embed|fieldset|figcaption|figure|font|footer|form|frame|frameset|h[1-6]|head|header|hr|html|i|iframe|img|input|ins|kbd|label|legend|li|link|main|map|mark|meta|meter|nav|noframes|noscript|object|ol|optgroup|option|output|p|param|picture|pre|progress|q|rp|rt|ruby|s|samp|script|section|select|small|source|span|strike|strong|style|sub|summary|sup|svg|table|tbody|td|template|textarea|tfoot|th|thead|time|title|tr|track|tt|u|ul|var|video|wbr)\b > ref html/tag
```

Comment:
I wonder if these should be in a `try` block, in case the user doesn't load html.kak, though that seems unlikely.

Reply:
Out of consistency, this `require-module` probably shouldn't be there, to keep in line with the bit above:

https://github.com/mawww/kakoune/blob/df7b33bc7b9da897d1d0127b54ae1e629d2334ff/rc/filetype/markdown.kak#L43-L54

Now it's possible that this loop *should* be calling `require-module` itself, but it's not really self-evident, so maybe we should take care of that in a subsequent issue/PR (or not).

Reply:
The `require-module html` needs to be there for `ref html/tag` to work. Sure, the `require-module` command that gets run when you type <code>```html</code> into a Markdown document is redundant but there does not seem to be any issue with running `require-module html` twice (the description for the `require-module` command is “ensures that <var>name</var> module has been loaded”).

We could replace `ref html/tag` with `group` but this results in poorer highlighting and makes debugging harder if the `markdown/tag` highlighter causes a false positive (I am not aware of such a case but you never know).